install.sh

```diff
@@ -139,25 +151,31 @@ ALLOWED="python3.10 \"$FOLDER/allowed.py\" -c \"$FOLDER/m269.json\""
 
 if [ $shell = "fish" ]
 then
-    FILE=~/.config/fish/config.fish
+    SHELL_CONFIG_FILE=~/.config/fish/config.fish
 else
-    FILE=~/.${shell}rc
+    SHELL_CONFIG_FILE=~/.${shell}rc
 fi
 
 if [ $shell = "csh" ] || [ $shell = "tcsh" ]
 then
-    echo "alias $COURSE '$M269.csh'" >> $FILE
-    echo "alias nb '$NB'" >> $FILE
-    echo "alias allowed '$ALLOWED'" >> $FILE
+    echo "alias $COURSE '$M269.csh'" >> $SHELL_CONFIG_FILE
+    echo "alias nb '$NB'" >> $SHELL_CONFIG_FILE
+    echo "alias allowed '$ALLOWED'" >> $SHELL_CONFIG_FILE
 else
     if [ $shell = "fish" ]
     then
-        echo "alias $COURSE='$M269.fish'" >> $FILE
+        echo "alias $COURSE='$M269.fish'" >> $SHELL_CONFIG_FILE
     else
-        echo "alias $COURSE='$M269'" >> $FILE
+        echo "alias $COURSE='$M269'" >> $SHELL_CONFIG_FILE
     fi
-    echo "alias nb='$NB'" >> $FILE
-    echo "alias allowed='$ALLOWED'" >> $FILE
+    echo "alias nb='$NB'" >> $SHELL_CONFIG_FILE
+    echo "alias allowed='$ALLOWED'" >> $SHELL_CONFIG_FILE
 fi
 
-echo "All done. Go to $SITE for further instructions."
\ No newline at end of file
+# Set variables in uninstall.sh
+echo "Setting variables into uninstall.sh ..."
+sed -i "14iFOLDER=$FOLDER" "$FOLDER/$UNINSTALL"
```

Comment:
I've only tried running `install.sh` in macOS Big Sur and didn't work. Gives and error on this and the next line and doesn't modify `uninstall.sh`.  
The installed sed doesn't seem to be GNU sed. The `-i` flag expects (after a space) an extension for the backup. Can be `''` to not have a backup. Probably you need to check what POSIX sed allows to be compatible across platforms.
However, both the macOS version and GNU sed expect a backslash after `14i` with the text to be inserted in the next line. Maybe you can insert the two lines in one single sed insert command?